### ISSUE TITLE
Errors on a char that doesn't have the reagent bank unlocked

### ISSUE THREAD

**Issue by cremor** [gsnerf/Baud-Bag#9]
I have a char that doesn't have the reagent bank unlocked. I get an error on that char:

On login/reload and when opening the bank:
```
1x BaudBag/ReagentBank.lua:53: attempt to index local 'bagSlot' (a nil value)
[string "@BaudBag/ReagentBank.lua"]:53: in function <BaudBag/ReagentBank.lua:41>
[string "=[C]"]: in function `BankBags_Update'
[string "@BaudBag/Bank.lua"]:114: in function `BaudBagBankBags_Update'
[string "@BaudBag/Bank.lua"]:81: in function `BaudBagBankBags_Initialize'
[string "@BaudBag/BaudBag.lua"]:74: in function `?'
[string "@BaudBag/BaudBag.lua"]:349: in function <BaudBag/BaudBag.lua:346>
```

When hovering over the bag bar in the Baud Bag options window:
```
1x BaudBag/widgets/buttons/BagButton.lua:253: attempt to index field '?' (a nil value)
[string "@BaudBag/widgets/buttons/BagButton.lua"]:253: in function <BaudBag/widgets/buttons/BagButton.lua:244>
```

What's also weird is that Baud Bag shows the reagent bank like I would have bought it. There is no indicator that it is locked and no option to buy it.

I'm using v10.0.5.1_beta_1

**Comment by gsnerf** [comment]
Hmm, it sounds like the "buy now" overlay may be broken after I moved those things to the reagent bank file. Will have a closer look. Does it still happen with v10.0.5.1 final?

**Comment by cremor** [comment]
The error in the options window seems to be fixed. The other ones are still there.
On login:
```
1x BaudBag/ReagentBank.lua:53: attempt to index local 'bagSlot' (a nil value)
[string "@BaudBag/ReagentBank.lua"]:53: in function <BaudBag/ReagentBank.lua:41>
[string "=[C]"]: in function `BankBags_Update'
[string "@BaudBag/Bank.lua"]:114: in function `BaudBagBankBags_Update'
[string "@BaudBag/Bank.lua"]:81: in function `BaudBagBankBags_Initialize'
[string "@BaudBag/BaudBag.lua"]:74: in function `?'
[string "@BaudBag/BaudBag.lua"]:349: in function <BaudBag/BaudBag.lua:346>
```

When opening the bank:
```
1x BaudBag/ReagentBank.lua:53: attempt to index local 'bagSlot' (a nil value)
[string "@BaudBag/ReagentBank.lua"]:53: in function <BaudBag/ReagentBank.lua:41>
[string "=[C]"]: in function `BankBags_Update'
[string "@BaudBag/Bank.lua"]:114: in function `BaudBagBankBags_Update'
[string "@BaudBag/Bank.lua"]:157: in function <BaudBag/Bank.lua:147>
[string "=[C]"]: in function `BankBags_UpdateContent'
[string "@BaudBag/Bank.lua"]:34: in function `?'
[string "@BaudBag/Bank.lua"]:55: in function `BaudBag_OnBankEvent'
[string "@BaudBag/BaudBag.lua"]:347: in function <BaudBag/BaudBag.lua:346>
```

**Comment by gsnerf** [comment]
Currently working on those. This seems to be broken since even before I started to extract the Reagent Bank for WotLK compatibility.

**Comment by gsnerf** [comment]
Hopefully fixed with v10.0.5.2

**Comment by cremor** [comment]
Can confirm, thanks!
